pkg/river/internal/value/value_test.go

```diff
@@ -9,33 +9,107 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-// TestInterfacePointerReceiver tests various cases where Go values which only
-// implement an interface for a pointer receiver are retained correctly
-// throughout values.
-func TestInterfacePointerReceiver(t *testing.T) {
-	t.Run("Encode", func(t *testing.T) {
-		pm := &pointerMarshaler{}
-
-		val := value.Encode(pm)
-		require.Equal(t, value.TypeString, val.Type())
-		require.Equal(t, "Hello, world!", val.Text())
-	})
+// TestEncodeKeyLookup tests where Go values are retained correctly
+// throughout values with a key lookup.
+func TestEncodeKeyLookup(t *testing.T) {
+	type Body struct {
+		Data pointerMarshaler `river:"data,attr"`
+	}
 
-	t.Run("From field", func(t *testing.T) {
-		type Body struct {
-			Data pointerMarshaler `river:"data,attr"`
-		}
+	tt := []struct {
+		name         string
+		encodeTarget any
+		key          string
+
+		expectBodyType  value.Type
+		expectKeyExists bool
+		expectKeyValue  value.Value
+		expectKeyType   value.Type
+	}{
+		{
```

Comment:
@erikbaranowski Can you add a test case here for getting a zero value from a River object?

For example, for `map[string]string{"data": ""}` the lookup should succeed and the resulting value should be the empty string. 

After that, I think this is good to merge.